lib/routes/routesUtils.js

```diff
@@ -244,7 +244,7 @@ const routesUtils = {
             }
             const partStart = parseInt(dataLocations[i].start, 10);
             const partSize = parseInt(dataLocations[i].size, 10);
-            if (partStart + partSize < begin) {
+            if (partStart + partSize <= begin) {
```

Comment:
This change has been merged into rel/6.2.1 branch. It is included to enable refactoring the same test updated in that PR. See https://github.com/scality/S3/commit/d7a568539e1fde90f8402b1a2399d72c093a8d19.